### PR TITLE
Prometheus Alerts for Fluentd and Rsyslog Log Collection

### PR DESCRIPTION
**Pull request by mburke5678**
https://jira.coreos.com/browse/PROD-825
https://jira.coreos.com/browse/LOG-258
https://jira.coreos.com/browse/LOG-9
https://jira.coreos.com/browse/OSDOCS-625

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@mburke5678 please make sure to check the labels.

**Comment by mburke5678** [comment]
Note to self: Do we need separate PR to add the Fluentd alerts to 4.1?

**Comment by mburke5678** [comment]
@nhosoi Am I correct here that your Jira https://jira.coreos.com/browse/LOG-258 adds alerts for Fluentd and Rsyslog that are shown in the Prometheus web console? PTAL

**Comment by richm** [comment]
> @nhosoi @richm
> Am I correct here that your Jira https://jira.coreos.com/browse/LOG-258 and https://jira.coreos.com/browse/LOG-9 add alerts for Fluentd and Rsyslog that are shown in the Prometheus web console? PTAL

Yes, you are correct.

**Comment by mburke5678** [comment]
@QiaolingTang Please take a look. 

@nhosoi @richm What does the user do when faced with these errors? Can they affect the queue buffer settings or max error values?

**Comment by richm** [comment]
> @QiaolingTang Please take a look.
> 
> @nhosoi @richm What does the user do when faced with these errors? Can they affect the queue buffer settings or max error values?

Yeah, I guess that's the next thing we need to do - write SOPs for every alert . . . the short answer is yes, there are things the user can do.

File: modules/efk-logging-fluentd-alerts.adoc [hunk 1]
```diff
@@ -2,44 +2,57 @@
...
-The contents of log files are printed out, starting with the oldest log. 
+[id="efk-logging-fluentd-alerts_{context}"]
+= About logging collector alerts
+
+The following alerts are generated by the logging collector and can be viewed on the *Alerts* tab of the Prometheus UI.
+
+If an alert is triggered, it appears on the *Monitoring* -> *Alerts* page of the {product-title} web console.
+
+.Fluentd Prometheus alerts
+|===
```
Comment (QiaolingTang) [inline modules/efk-logging-fluentd-alerts.adoc]: @mburke5678 The fluentd alerts are in https://github.com/openshift/cluster-logging-operator/blob/master/files/fluentd/fluentd_prometheus_alerts.yaml.

File: modules/efk-logging-fluentd-alerts.adoc [hunk 2]
```diff
@@ -2,44 +2,57 @@
...
+
+|`fluentdNumErrors`
+|Fluentd issues more than the specified number of errors, default 10.
+|`Fluentd reports high number of errors`
+
+|===
+
+.Rsyslog Prometheus alerts
+|===
+|Alert |Description |Example
```
Comment (QiaolingTang) [inline modules/efk-logging-fluentd-alerts.adoc]: Alert rules for rsyslog: https://github.com/openshift/cluster-logging-operator/blob/master/files/rsyslog/rsyslog_prometheus_alerts.yaml

**Comment by mburke5678** [comment]
@QialoingTang I took the alert names and messages from the linked document and added them to the docs. How does this look? 
Do you have a 4.2 console where I can see the alerts?

**Comment by mburke5678** [comment]
@QiaolingTang I took the alert names and messages from the linked document and added them to the docs. How does this look?
Do you have a 4.2 console where I can see the alerts?

**Comment by QiaolingTang** [comment]
@mburke5678 4.2 env information is sent to your Slack, please check.

**Comment by QiaolingTang** [comment]
LGTM, thanks.

**Comment by mburke5678** [comment]
Moving to 4.3. Removed references to Rsyslog

https://github.com/openshift/openshift-docs/pull/16597

**Pull request merged**